test/memspaces/memspace_highest_bandwidth.cpp

```diff
@@ -12,13 +12,23 @@
 static bool canQueryBandwidth(size_t nodeId) {
     hwloc_topology_t topology = nullptr;
     int ret = hwloc_topology_init(&topology);
-    UT_ASSERTeq(ret, 0);
+
+    if (!GTEST_OUT_EQ(ret, 0)) {
```

Comment:
Can the `EXPECT_EQ` macro be used instead? What is the advantage of using the `GTEST_OUT_EQ` macro over `EXPECT_EQ`?

Reply:
I have tried EXPECT_EQ() and it interferes with GTEST_SKIP(): although the tests are labeled as ,,skipped" in the fixture, they are executed. For example, failed ASSERT_NE(sth, nullptr) are outputted, because sth has been never initialized since the fixture stopped the SetUp() execution after GTEST_SKIP(). 

 ADD_FAILURE() gives similar result to EXPECT_EQ() (tests are eventually run).

Therefore I have decided to display the message and to manipulate the program execution only in the given fixture.

Reply:
If we need these GTEST checks, then I'm ok with that. Please, consider if we could get rid of the GTEST macros in this function and ie. return a `std::pair<bool, std::string>` with a message to pass to GTEST_SKIP instead. No need for introducing new workaround macros if this could work.